sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -1094,49 +1094,64 @@ func getIPAndPort(address *sensor.NetworkAddress) net.NetworkPeerID {
 	return tuple
 }
 
+func getConnection(conn *sensor.NetworkConnection) *connection {
+	var incoming bool
+	switch conn.Role {
+	case sensor.ClientServerRole_ROLE_SERVER:
+		incoming = true
+	case sensor.ClientServerRole_ROLE_CLIENT:
+		incoming = false
+	default:
+		return nil
+	}
+
+	remote := net.NumericEndpoint{
+		IPAndPort: getIPAndPort(conn.GetRemoteAddress()),
+		L4Proto:   net.L4ProtoFromProtobuf(conn.GetProtocol()),
+	}
+	local := getIPAndPort(conn.GetLocalAddress())
+
+	// Special handling for UDP ports - role reported by collector may be unreliable, so look at which port is more
+	// likely to be ephemeral. In case a port is set to 0, collector couldn't retrieve this value, we assume the
+	// connection works in the direction opposite of this port.
+	if remote.L4Proto == net.UDP {
+		incoming = netutil.IsEphemeralPort(remote.IPAndPort.Port) > netutil.IsEphemeralPort(local.Port)
+	}
+
+	return &connection{
+		local:       local,
+		remote:      remote,
+		containerID: conn.GetContainerId(),
```

Comment:
I suppose this container is either the receiving or sending end of a connection, right?

Reply:
I believe it's the container the `local` side is on, but the code in collector is a bit of a tangled mess with calls to Falco, I might be wrong.